pallets/staking/src/lib.rs

```diff
@@ -1964,12 +1999,21 @@ decl_module! {
             let ledger = Self::ledger(&controller).ok_or(Error::<T>::NotController)?;
             let stash = &ledger.stash;
 
-            ensure!(ledger.active >= <MinimumBondThreshold<T>>::get(), Error::<T>::InsufficientValue);
-            // Polymesh-Note - It is used to check whether the passed commission is <= commission cap.
-            ensure!(prefs.commission <= Self::validator_commission_cap(), Error::<T>::InvalidValidatorCommission);
+            // Polymesh-Note - Make sure stash has valid permissioned identity.
+            if let Some(id) = <Identity<T>>::get_identity(stash) {
+                // Ensure stash's identity is be permissioned.
+                let id_pref = Self::permissioned_identity(id)
+                    .ok_or_else(|| Error::<T>::StashIdentityNotPermissioned)?;
+                // Ensure identity doesn't run more validators than the intended count.
+                ensure!(id_pref.running_count < id_pref.intended_count, Error::<T>::HitIntendedValidatorCount);
+                ensure!(ledger.active >= <MinimumBondThreshold<T>>::get(), Error::<T>::InsufficientValue);
+                // Ensures that the passed commission is within the cap.
+                ensure!(prefs.commission <= Self::validator_commission_cap(), Error::<T>::InvalidValidatorCommission);
+                PermissionedIdentity::insert(id, PermissionedIdentityPrefs {intended_count: id_pref.intended_count, running_count: id_pref.running_count + 1});
```

Comment:
Do we need to account for the same controller calling validate twice? i.e. check that the origin is not already in the `Validators` list and reject if it is?

Reply:
I don't think so we need it, It doesn't really affect anything.

Reply:
btw it will be more performant to make `id_pref` mutable and increment the `running_count` there instead of making a new object here.